### PR TITLE
TST: add test for .loc indexing Index type preservation

### LINKED ISSUE #15166: BUG: .loc indexing not preserving Index type
#### 

```python
test = pd.DataFrame(data=np.arange(2,22,2), 
             index=pd.MultiIndex(levels=[pd.CategoricalIndex(["a", "b"]), range(10)],
                                 labels=[[0]*5 + [1]*5, range(10)],
                                 names = ["Index1", "Index2"]))
test.index.levels[0]
# CategoricalIndex([u'a', u'b'], categories=[u'a', u'b'], ordered=False, name=u'Index1', dtype='category')
test.loc[["a"]].index.levels[0]
# Index([u'a'], dtype='object', name=u'Index1')
```
#### Problem description
When selecting data through a categorical index, the categorical index is lost.

#### Expected Output

```
test.loc[["a"]].index.levels[0]
# CategoricalIndex([u'a'], categories=[u'a', u'b'], ordered=False, name=u'Index1', dtype='category')
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.10.final.0
python-bits: 64
OS: Linux
OS-release: 2.6.32-431.11.2.el6.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: en_GB.utf8
LANG: en_GB.utf8
LOCALE: None.None

pandas: 0.19.2
nose: 1.3.7
pip: 9.0.1
setuptools: 32.3.1
Cython: None
numpy: 1.11.3
scipy: 0.18.1
statsmodels: None
xarray: None
IPython: 5.1.0
sphinx: 1.5.1
patsy: None
dateutil: 2.6.0
pytz: 2016.10
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: 1.5.3
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.9999999
httplib2: None
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.9.4
boto: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by ganevgv**
- [x] closes #15166
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/test_indexing.py [hunk 1]
```diff
@@ -1944,6 +1944,29 @@ def test_set_value_resize(self, float_frame):
         with pytest.raises(ValueError, match=msg):
             res3._set_value("foobar", "baz", "sam")
 
+    def test_loc_indexing_preserves_index_dtype(self):
```
Comment (mroeschke) [inline pandas/tests/frame/test_indexing.py]: Can you mention categorical in the title?

Comment (ganevgv) [reply pandas/tests/frame/test_indexing.py]: Thanks, @mroeschke - done.

**Review by mroeschke — APPROVED** [review]

File: pandas/tests/frame/test_indexing.py [hunk 2]
```diff
@@ -1944,6 +1944,29 @@ def test_set_value_resize(self, float_frame):
...
+        df = DataFrame(
+            data=np.arange(2, 22, 2),
+            index=pd.MultiIndex(
+                levels=[pd.CategoricalIndex(["a", "b"]), range(10)],
+                codes=[[0] * 5 + [1] * 5, range(10)],
+                names=["Index1", "Index2"],
+            ),
+        )
+
+        result_1 = df.index.levels[0]
```
Comment (jreback) [inline pandas/tests/frame/test_indexing.py]: can you write this like

expected=

result=
tm.assert_index_equal....

result=
tm.assert_index_equal....

File: pandas/tests/frame/test_indexing.py [hunk 3]
```diff
@@ -1944,6 +1944,29 @@ def test_set_value_resize(self, float_frame):
         with pytest.raises(ValueError, match=msg):
             res3._set_value("foobar", "baz", "sam")
 
+    def test_loc_indexing_preserves_index_category_dtype(self):
```
Comment (jreback) [inline pandas/tests/frame/test_indexing.py]: can you move this to: TestDataFrameIndexingCategorical

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/test_indexing.py (hunk 2 above)
Comment (ganevgv) [reply pandas/tests/frame/test_indexing.py]: Thanks, @jreback - done.

File: pandas/tests/frame/test_indexing.py (hunk 3 above)
Comment (ganevgv) [reply pandas/tests/frame/test_indexing.py]: Thanks, @jreback - unfortunately, I couldn't find `tests/frame/indexing/test_...`, that's why I put the test in `tests/frame/test_indexing.py`.

However, there seems to be a slight inconsistency in the `tests/frame/` and `tests/series/` structure as there's a `indexing` subfolder in `series` but not in `frame`.

Comment (jreback) [reply pandas/tests/frame/test_indexing.py]: this is a class name *in* the current file

> However, there seems to be a slight inconsistency in the tests/frame/ and tests/series/ structure as there's a indexing subfolder in series but not in frame.

yes I know, there is an issue about fixing this (happy to have a PR), but orthogonal to this one.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/test_indexing.py (hunk 3 above)
Comment (ganevgv) [reply pandas/tests/frame/test_indexing.py]: Ah, I see - moved the test there. Thanks

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ganevgv

**Pull request merged**